packages/core/core/src/services/document-service/repository.ts

```diff
@@ -254,20 +262,21 @@ export const createContentTypeRepository: RepositoryFactoryMethod = (uid) => {
   async function unpublish(opts = {} as any) {
     const { documentId, ...params } = opts;
 
-    const queryParams = await async.pipe(
+    const query = await async.pipe(
       i18n.defaultLocale(contentType),
-      i18n.multiLocaleToLookup(contentType)
+      i18n.multiLocaleToLookup(contentType),
+      transformParamsToQuery(uid),
+      (query) => assoc('where', { ...query.where, documentId, publishedAt: { $ne: null } }, query)
     )(params);
 
-    const { deletedEntries } = await deleteDocument({
-      ...params,
-      documentId,
-      lookup: { ...queryParams?.lookup, publishedAt: { $ne: null } },
-    });
+    // Delete all published versions
+    const versionsToDelete = await strapi.db?.query(uid).findMany(query);
```

Comment:
didn't you want to ensure populating data for webhooks ?

Reply:
the payload webhooks used does not come from here, but rather an async findOne request, we can discuss alternatives as there are edge cases that are not covered by doing so (delete & unpublish)

Reply:
well, maybe it makes sense to deep populate those versions too in the unpublish scenario 🤔 , so we always have that data populated, wdyt?

Reply:
I think it's worth doing a quick benchmark on how others manage webhooks for the "data" loss events. 

I don't have a strong opinion on it as long as we can have something consistent

Reply:
alright :) will check it